twake/backend/node/src/services/tags/services/tags.ts

```diff
@@ -0,0 +1,76 @@
+import { TwakeServiceProvider, Initializable } from "src/core/platform/framework/api";
+import Repository from "src/core/platform/services/database/services/orm/repository/repository";
+import { Tags, TagsType, TagsPrimaryKey } from "../entities";
+import gr from "../../global-resolver";
+import {
+  DeleteResult,
+  ExecutionContext,
+  ListResult,
+  OperationType,
+  SaveResult,
+} from "../../../core/platform/framework/api/crud-service";
+import { plainToClass } from "class-transformer";
+import { createTagEntity, TYPE } from "../entities/tags";
+import { logger } from "../../../core/platform/framework";
+
+export class TagsService implements TwakeServiceProvider, Initializable {
+  version: "1";
+  repository: Repository<Tags>;
+
+  async init(): Promise<this> {
+    try {
+      this.repository = await gr.database.getRepository<Tags>(TYPE, Tags);
+    } catch (err) {
+      console.log(err);
+      logger.error("Error while initializing tags service");
+    }
+    return this;
+  }
+
+  async get(pk: TagsPrimaryKey): Promise<Tags> {
+    return await this.repository.findOne({ company_id: pk.company_id, tag_id: pk.tag_id });
+  }
+
+  async save(tag: Tags, context: ExecutionContext): Promise<SaveResult<Tags>> {
+    /*
+    //Creation
+    {
+      name: "Tag principal",
+      colour: "#F0AF23",
+      tag_id: undefined
+    }
+    -> tag_id: "super_tag"
+
+    //Update
+    {
+      name: "Tag principal",
+      colour: "#F0AF23",
+      tag_id: "super_tag"
+    }
+    -> tag_id: "super_tag"
+    */
+    const tagToSave = createTagEntity(tag);
+    tagToSave.tag_id = tag.name.split(" ").join("_");
```

Comment:
To create a good tag id, use this instead: `name.toLocaleLowerCase().replace(/[^a-z0-9]/gm, "_")`